Framework/Geometry/src/Instrument/CompAssembly.cpp

```diff
@@ -344,7 +345,8 @@ CompAssembly::getComponentByName(const std::string &cname, int nlevels) const {
   } // while-end
 
   // If we have reached here then the search failed
-  return boost::shared_ptr<const IComponent>();
+  throw std::runtime_error("Error when trying to find the component " + cname +
+                           " , this component may not exist. \n");
 }
```

Comment:
One unit test requires invalid argument exception to be thrown if the component is not found (not runtime error).